lib/json/ld/context.rb

```diff
@@ -791,15 +792,15 @@ def from_vocabulary(graph)
         (statements[statement.subject] ||= []) << statement
 
         # Keep track of predicate ranges
-        if [RDF::RDFS.range, RDF::SCHEMA.rangeIncludes].include?(statement.predicate) 
+        if [RDF::RDFS.range, RDF::SCHEMA.rangeIncludes].include?(statement.predicate)
           (ranges[statement.subject] ||= []) << statement.object
         end
       end
 
       # Add term definitions for each class and property not in vocab, and
       # for those properties having an object range
       statements.each do |subject, values|
-        types = values.select {|v| v.predicate == RDF.type}.map(&:object)
+        types = values.each_with_object([]) { |v, memo| memo << v.object if v.predicate == RDF.type }
```

Comment:
I wasn't familiar with this use of `#each_with_object`, interesting.

Reply:
Here we avoid the double Array allocation (`.select{...}.map`). Lots of small changes like this, big impact.